setup.py

```diff
@@ -24,46 +28,141 @@ def load_version_info(version_file):
         return json.load(f)
 
 
-try:
-    version_info = load_version_info(VERSION_FILE_LOCAL)
-except FileNotFoundError:
-    print("version_local.json not found. Default to dev build")
-    version_info = load_version_info(VERSION_FILE)
+def get_package_version():
+    """Load and return the package version from version files."""
+    try:
+        version_info = load_version_info(VERSION_FILE_LOCAL)
+    except FileNotFoundError:
+        print("version_local.json not found. Default to dev build")
+        version_info = load_version_info(VERSION_FILE)
 
-PACKAGE_VERSION = version_info["package-version"]
-print(f"Using PACKAGE_VERSION: '{PACKAGE_VERSION}'")
-
-with open(os.path.join(REPO_ROOT, "README.md"), "rt") as f:
-    README = f.read()
-
-packages = find_namespace_packages(
-    include=[
-        "iree.turbine",
-        "iree.turbine.*",
-    ],
-)
-
-print("Found packages:", packages)
-
-# Lookup version pins from requirements files.
-requirement_pins = {}
+    package_version = version_info["package-version"]
+    print(f"Using PACKAGE_VERSION: '{package_version}'")
+    return package_version
 
 
 def load_requirement_pins(requirements_file: str):
+    """Load version pins from requirements file."""
     with open(Path(THIS_DIR) / requirements_file, "rt") as f:
         lines = f.readlines()
     pin_pairs = [line.strip().split("==") for line in lines if "==" in line]
-    requirement_pins.update(dict(pin_pairs))
-
-
-load_requirement_pins("requirements.txt")
+    return dict(pin_pairs)
 
 
-def get_version_spec(dep: str):
+def get_version_spec(dep: str, requirement_pins: dict):
+    """Get version specification for a dependency."""
     if dep in requirement_pins:
         return f">={requirement_pins[dep]}"
-    else:
-        return ""
+    return ""
+
+
+def check_nanobind_available():
+    """Check if nanobind is available in the current Python environment."""
+    try:
+        import nanobind
+
+        return True
+    except ImportError:
+        return False
+
+
+def check_torch_cuda_available():
+    """Check if PyTorch has CUDA support."""
+    try:
+        import torch
+
+        return torch.cuda.is_available()
+    except ImportError:
+        return False
+
+
+def setup_cmake_environment():
+    """Setup CMake environment with system Python."""
+    python_executable = sys.executable
+    cmake_env = os.environ.copy()
+    cmake_args = []
+
+    print(f"Using Python: {python_executable}")
+
+    return python_executable, cmake_env, cmake_args
```

Comment:
I don't see much value in this function as it exists now. I was confused seeing `python_executable` below instead of `sys.executable`, so the indirection / encapsulation isn't helping readability.